frontend/packages/ceph-storage-plugin/src/components/volume-snapshot/volume-snapshot.tsx

```diff
@@ -0,0 +1,233 @@
+            <dl>
+              <dt>Name</dt>
+              <dd>{_.get(resource, 'metadata.name')}</dd>
+              <dt>Date</dt>
+              <dd>{_.get(resource, 'metadata.creationTimestamp')}</dd>
+              <dt>Status</dt>
+              <dd>{_.get(resource, 'status.readyToUse') ? 'Ready' : 'Not Ready'}</dd>
+              <dt>Size</dt>
+              <dd>
+                {_.get(resource, 'status.restoreSize')
+                  ? _.get(resource, 'status.restoreSize')
+                  : 'No Data'}
+              </dd>
+              <dt>Namespace</dt>
+              <dd>
+                <ResourceLink kind="Namespace" />
+                {_.get(resource, 'metadata.namespace')}
+              </dd>
+              <dt>Annotations</dt>
+              <dd>None</dd>
+              <dt>Labels</dt>
+              <dd>None</dd>
+            </dl>
+          </div>
+          <div className="col-sm-6">
+            <dl>
+              <dt>API Version</dt>
+              <dd>{_.get(resource, 'apiVersion')}</dd>
+              <dt>Persistent Volume Claim</dt>
+              <dd>
+                <ResourceLink kind="PersistentVolumeClaim" />
+                {_.get(resource, 'spec.source.name')}
+              </dd>
+            </dl>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const VolumeSnapshotTableRow = ({ obj, index, key, style }) => {
+  return (
+    <TableRow id={obj.metadata.uid} index={index} trKey={key} style={style}>
+      <TableData className={snapshotTableColumnClasses[0]}>
+        <VolumeSnapshotLink
+          name={_.get(obj, 'metadata.name')}
+          namespace={_.get(obj, 'metadata.namespace')}
+          pvc={_.get(obj, 'spec.source.name')}
+          details={VolumeSnapshotDetails}
+        />
+      </TableData>
+      <TableData className={snapshotTableColumnClasses[1]}>
+        {_.get(obj, 'metadata.creationTimestamp')}
+      </TableData>
+      <TableData className={snapshotTableColumnClasses[2]}>
+        {_.get(obj, 'status.readyToUse') ? 'Ready' : 'Not Ready'}
+      </TableData>
+      <TableData className={snapshotTableColumnClasses[3]}>
+        {_.get(obj, 'status.restoreSize') ? _.get(obj, 'status.restoreSize') : 'No Data'}
+      </TableData>
+      <TableData className={snapshotTableColumnClasses[4]}>None</TableData>
+      <TableData className={snapshotTableColumnClasses[5]}>
+        <ResourceKebab actions={snapshotMenuActions} kind={kind} resource={obj} />
+      </TableData>
+    </TableRow>
+  );
+};
+VolumeSnapshotTableRow.displayName = 'SnapshotTableRow';
+
+export const VolumeSnapshotList: React.FC = (props) => (
+  <Table
+    {...props}
+    aria-label="Volume Snapshot"
+    Header={VolumeSnapshotTableHeader}
+    Row={VolumeSnapshotTableRow}
+    virtualize
+  />
+);
+
+export const VolumeSnapshotLink = (props) => {
+  const { name, namespace, pvc } = props;
+  const path = `/k8s/ns/${namespace}/persistentvolumeclaims/${pvc}/volumesnapshots/${name}`;
```

Comment:
Why do we need a special path for volume snapshots? It should follow the standard URL patterns so it renders the same everywhere (i.e. from the Search page)

Reply:
Attaching the Design: https://marvelapp.com/100ah4f4/screen/64180287
Other resources have independent views and tab, since the snapshot is not listed as an independent tab and are paired with particular PVCs, that's why created a separate path. Also, according to the design we needed different breadcrumbs than at standard view. 
@yuvalgalanti @cloudbehl  @spadgett should we keep it like other resource or keep it same as design?

Reply:
You should have a standalone resource page for snapshots and just include them as a tab in the PVC. We have many of places in console you can follow to do this (e.g., the ReplicaSet page has a Pods tab). Follow the established patterns. This really isn't a special case, and you should be able to implement the design with no trouble.